crypto/secp256k1/secp256k1.go

```diff
@@ -29,10 +29,15 @@ func NewPublicKey(x, y [32]byte) *PublicKey {
 
 // NewPublicKeyFromBigInt returns a new public key from the given (x, y) coordinates
 func NewPublicKeyFromBigInt(x, y *big.Int) *PublicKey {
-	var xb, yb [32]byte
-	copy(xb[:], x.Bytes())
-	copy(yb[:], y.Bytes())
-	return NewPublicKey(xb, yb)
+	const ptSize = 32
+	var xArray, yArray [ptSize]byte
+	xSlice := x.Bytes()
+	ySlice := y.Bytes()
+	// Copying from a big-endian slice into a big-endian array, so we want padding bytes
+	// on the left if the slice is shorter than the array.
+	copy(xArray[ptSize-len(xSlice):], xSlice)
+	copy(yArray[ptSize-len(ySlice):], ySlice)
+	return NewPublicKey(xArray, yArray)
```

Comment:
This was the code change that fixes the problem.

Reply:
awesome, makes sense!